Source/Core/Common/Matrix.h

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+
+// Tiny matrix/vector library.
+// Used for things like Free-Look in the gfx backend.
+
+union Vec3
+{
+  Vec3() {}
+  Vec3(float _x, float _y, float _z) : data{_x, _y, _z} {}
+
+  Vec3& operator+=(const Vec3& rhs)
+  {
+    x += rhs.x;
+    y += rhs.y;
+    z += rhs.z;
+    return *this;
+  }
+
+  std::array<float, 3> data = {};
+
+  struct
+  {
+    float x;
+    float y;
+    float z;
+  };
+};
+
+inline Vec3 operator+(Vec3 lhs, const Vec3& rhs)
+{
+  return lhs += rhs;
+}
+
+static_assert(sizeof(Vec3) == sizeof(float) * 3, "Wrong size");
```

Comment:
This seems kind of unnecessary, but if it's going to be kept, it should probably have a better compilation error message than "Wrong size"

Reply:
I just removed it.